playwright/goal-sharing-feature.spec.ts

```diff
@@ -0,0 +1,66 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { test, expect, Page } from "@playwright/test";
+import { shareGoalPrivately } from "./lib/testUtils";
+
+test.describe.configure({ mode: "serial" });
+
+test.describe("Goal Sharing Feature", () => {
+  test.setTimeout(40000);
+  let userOneContext;
+  let userTwoContext;
+  let userOnePage: Page;
+  let userTwoPage: Page;
+  let link: string;
+
+  test.beforeAll(async ({ browser }) => {
+    userOneContext = await browser.newContext({
+      storageState: "playwright/userOnboarding.json",
+    });
+    userTwoContext = await browser.newContext({
+      storageState: "playwright/userOnboarding.json",
+    });
+
+    userTwoPage = await userTwoContext.newPage();
+    userOnePage = await userOneContext.newPage();
+  });
+
+  test("add contact in user 1", async () => {
+    await userOnePage.goto("http://localhost:3000/");
+    await userOnePage.getByRole("button", { name: "Goals" }).click();
+    await shareGoalPrivately(userOnePage);
+    await userOnePage.getByPlaceholder("Name").click();
+    await userOnePage.getByPlaceholder("Name").fill("User 2");
+    await userOnePage.getByRole("button", { name: "add contact Share invitation" }).click();
+    await userOnePage.waitForTimeout(5000);
```

Comment:
adding timeout is not a good approach, can you check if there is any function that we can use which waits for the function execution instead of using timeout